lib/vm.js

```diff
@@ -390,17 +391,26 @@ function compileFunction(code, params, options = {}) {
     }
   });
 
-  return _compileFunction(
+  const fn = _compileFunction(
     code,
     filename,
     lineOffset,
     columnOffset,
     cachedData,
-    produceCachedData,
     parsingContext,
     contextExtensions,
     params
   );
+
+  if (produceCachedData) {
+    const cache = createCacheForFunction(fn);
+    fn.cachedDataProduced = cache !== undefined;
+    if (fn.cachedDataProduced) {
+      fn.cachedData = cache;
```

Comment:
i can imagine this making people need to delete the prop after grabbing it if they want to pass the resulting function around.

Reply:
This has been out since v10, though this has not been tested nor documented. We'll probably need a deprecation cycle if we want to replace this with something else.

Reply:
i think we also need to allow this to run after the function has been called. we might need to do something like `vm.createCodeCacheForFunction(fn)`

Reply:
So @devsnek and @jdalton support exposing this as part of the `vm` API, unless I'm wrong? That is complete fine by me, I was actually planning of proposing that myself.

As of removing the cache production part inside `vm.compileFunction`, I honestly don't know. Idk if going through the deprecation cycle is worth it, but if everyone else feels it is a great idea to drop the prop, I have no real objections backed with logic.